### ISSUE TITLE
Ensemble Builder fails with Timeout Exception

### ISSUE THREAD

**Issue by heartnetkung** [automl/auto-sklearn#609]
I've encountered an ensemble error with large dataset(7 million rows, 23 columns) making the final ensemble unusable. The log says that there is a time out happening with pynisher.limit_function_call.TimeoutException. The model result has bad accuracy and contains only an ensemble of 2 models which is weird since the log file shows that there are at least 10 models doing better than the average and training loss are quite good.

Here is the result of model.get_models_with_weights()
```
[(0.86,
  SimpleRegressionPipeline({'regressor:random_forest:min_samples_leaf': 11, 'imputation:strategy': 'median', 'regressor:random_forest:max_leaf_nodes': 'None', 'regressor:__choice__': 'random_forest', 'regressor:random_forest:n_estimators': 100, 'rescaling:__choice__': 'none', 'preprocessor:pca:whiten': 'False', 'regressor:random_forest:max_depth': 'None', 'regressor:random_forest:bootstrap': 'False', 'preprocessor:pca:keep_variance': 0.903759669755001, 'regressor:random_forest:max_features': 0.9303289050883723, 'regressor:random_forest:min_samples_split': 11, 'categorical_encoding:__choice__': 'one_hot_encoding', 'regressor:random_forest:min_weight_fraction_leaf': 0.0, 'regressor:random_forest:criterion': 'friedman_mse', 'preprocessor:__choice__': 'pca', 'regressor:random_forest:min_impurity_decrease': 0.0, 'categorical_encoding:one_hot_encoding:use_minimum_fraction': 'False'},
  dataset_properties={
    'task': 4,
    'signed': False,
    'multiclass': False,
    'target_type': 'regression',
    'multilabel': False,
    'sparse': False})),
 (0.14,
  SimpleRegressionPipeline({'regressor:random_forest:min_samples_leaf': 19, 'imputation:strategy': 'mean', 'regressor:random_forest:max_leaf_nodes': 'None', 'regressor:__choice__': 'random_forest', 'regressor:random_forest:n_estimators': 100, 'rescaling:__choice__': 'normalize', 'preprocessor:pca:whiten': 'True', 'regressor:random_forest:max_depth': 'None', 'regressor:random_forest:bootstrap': 'True', 'preprocessor:pca:keep_variance': 0.5301008473715315, 'regressor:random_forest:max_features': 0.5985889681620831, 'regressor:random_forest:min_samples_split': 8, 'categorical_encoding:__choice__': 'one_hot_encoding', 'regressor:random_forest:min_weight_fraction_leaf': 0.0, 'regressor:random_forest:criterion': 'friedman_mse', 'preprocessor:__choice__': 'pca', 'regressor:random_forest:min_impurity_decrease': 0.0, 'categorical_encoding:one_hot_encoding:use_minimum_fraction': 'False'},
  dataset_properties={
    'task': 4,
    'signed': False,
    'multiclass': False,
    'target_type': 'regression',
    'multilabel': False,
    'sparse': False}))]

```

Here is the tail of the log file.
```
[INFO] [2019-01-05 21:21:20,739:smac.stats.stats.Stats] ##########################################################
[INFO] [2019-01-05 21:21:20,739:smac.stats.stats.Stats] Statistics:
[INFO] [2019-01-05 21:21:20,739:smac.stats.stats.Stats] #Incumbent changed: 3
[INFO] [2019-01-05 21:21:20,739:smac.stats.stats.Stats] #Target algorithm runs: 25 / inf
[INFO] [2019-01-05 21:21:20,739:smac.stats.stats.Stats] Used wallclock time: 21560.75 / 21560.65 sec 
[INFO] [2019-01-05 21:21:20,739:smac.stats.stats.Stats] Used target algorithm runtime: 21525.49 / inf sec
[DEBUG] [2019-01-05 21:21:20,740:smac.stats.stats.Stats] Debug Statistics:
[DEBUG] [2019-01-05 21:21:20,740:smac.stats.stats.Stats] Average Configurations per Intensify: 2.00
[DEBUG] [2019-01-05 21:21:20,740:smac.stats.stats.Stats] Exponential Moving Average of Configurations per Intensify: 2.00
[INFO] [2019-01-05 21:21:20,740:smac.stats.stats.Stats] ##########################################################
[INFO] [2019-01-05 21:21:20,740:smac.facade.smac_facade.SMAC] Final Incumbent: Configuration:
  categorical_encoding:__choice__, Value: 'one_hot_encoding'
  categorical_encoding:one_hot_encoding:use_minimum_fraction, Value: 'False'
  imputation:strategy, Value: 'most_frequent'
  preprocessor:__choice__, Value: 'feature_agglomeration'
  preprocessor:feature_agglomeration:affinity, Value: 'manhattan'
  preprocessor:feature_agglomeration:linkage, Value: 'average'
  preprocessor:feature_agglomeration:n_clusters, Value: 103
  preprocessor:feature_agglomeration:pooling_func, Value: 'median'
  regressor:__choice__, Value: 'ridge_regression'
  regressor:ridge_regression:alpha, Value: 1.4467302287296093
  regressor:ridge_regression:fit_intercept, Constant: 'True'
  regressor:ridge_regression:tol, Value: 0.0029020092462586676
  rescaling:__choice__, Value: 'robust_scaler'
  rescaling:robust_scaler:q_max, Value: 0.8385049904773149
  rescaling:robust_scaler:q_min, Value: 0.17362159033588861

[DEBUG] [2019-01-05 21:21:21,911:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 20.003785
[DEBUG] [2019-01-05 21:21:21,912:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:21,912:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:21,912:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:22,119:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:24,127:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 17.787702
[DEBUG] [2019-01-05 21:21:24,128:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:24,128:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:24,128:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:24,335:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:26,344:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 15.570773
[DEBUG] [2019-01-05 21:21:26,345:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:26,345:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:26,345:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:26,551:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:28,560:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 13.354989
[DEBUG] [2019-01-05 21:21:28,560:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:28,561:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:28,561:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:28,767:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:30,775:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 11.139978
[DEBUG] [2019-01-05 21:21:30,775:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:30,776:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:30,776:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:30,978:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:32,986:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 8.929102
[DEBUG] [2019-01-05 21:21:32,986:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:32,987:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:32,987:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:33,187:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:35,195:EnsembleBuilder(1):bikeshare] Starting iteration 12, time left: 6.720284
[DEBUG] [2019-01-05 21:21:35,195:EnsembleBuilder(1):bikeshare] Read ensemble data set predictions
[DEBUG] [2019-01-05 21:21:35,196:EnsembleBuilder(1):bikeshare] Done reading 0 new prediction files. Loaded 12 predictions in total.
[DEBUG] [2019-01-05 21:21:35,196:EnsembleBuilder(1):bikeshare] Use -0.000001 as dummy score
[DEBUG] [2019-01-05 21:21:35,405:EnsembleBuilder(1):bikeshare] No new model predictions selected -- skip ensemble building -- current performance: 0.303054
[DEBUG] [2019-01-05 21:21:36,335:EnsembleBuilder(1):bikeshare] signal handler: 14
[DEBUG] [2019-01-05 21:21:36,335:EnsembleBuilder(1):bikeshare] timeout
[DEBUG] [2019-01-05 21:21:36,335:EnsembleBuilder(1):bikeshare] return value: (None, <class 'pynisher.limit_function_call.TimeoutException'>)
```

Here is the full logfile 
[AutoML(1):bikeshare.log](https://github.com/automl/auto-sklearn/files/2731914/AutoML.1.bikeshare.log)

**Comment by heartnetkung** [comment]
Here is the minimal code to run
```
def model1():
    feature = raw_df.drop(['holiday','station', 'min_date','date','log_demand'],axis = 1)
    label = raw_df['log_demand']
    feature_types = ['numerical'] + (['numerical']*10) +(['categorical','numerical','numerical']) + (['numerical'] * 2)
    X_train, X_test, y_train, y_test = \
        sklearn.model_selection.train_test_split(feature, label, test_size =0.15, shuffle=True, random_state = 42)

    automl = autosklearn.regression.AutoSklearnRegressor(
        time_left_for_this_task=3600*10,
        per_run_time_limit=3600,
        ml_memory_limit = 20000,
        ensemble_memory_limit=20000,
        output_folder='output',
        tmp_folder='temp',
        delete_output_folder_after_terminate=False,
        delete_tmp_folder_after_terminate = False,
        include_estimators=["random_forest", 'ridge_regression', 'extra_trees', 'libsvm_svr', 'decision_tree', 'ard_regression', 'sgd', 'adaboost']
    )
    automl.fit(X_train, y_train, dataset_name='bikeshare',
               feat_type=feature_types)

    print(automl.show_models())
    dump(automl, 'model1.joblib')
```

**Comment by heartnetkung** [comment]
My apology for this issue. Nothing is wrong with the model. It's just that the accuracy is bad due to bad features. The trajectory in the log file says it all.
